libs/openFrameworks/gl/ofShader.h

```diff
@@ -127,6 +127,38 @@ class ofShader {
 	
 	void setUniforms(const ofParameterGroup & parameters) const;
 
+    /// Changes a numerical define value
+    ///
+    /// Allows to change a define in the shader declared like
+    ///
+    /// #define SOME_FEATURE 1
+    ///
+    /// calling setDefineConstant("SOME_FEATURE", 0);
+    ///
+    /// Be aware that this recompiles the shader which can be
+    /// slow if done often. This is only recommendable for
+    /// values that don't change very often and if they
+    /// were a uniform would introduce runtime evaluated
+    /// conditionals or similar which would slow down the shader
+    void setDefineConstant(const string & name, float value);
+    void setDefineConstant(const string & name, int value);
+    void setDefineConstant(const string & name, bool value);
+
+
+    /// Changes a numerical define value
+    ///
+    /// Allows to change a define in the shader declared like
+    ///
+    /// const float somevalue=10;
+    ///
+    /// calling setConstant1f("somevalue", 20.f);
+    ///
+    /// Be aware that this recompiles the shader which can be
+    /// slow if done often.
+    void setConstant1f(const string & name, float value);
```

Comment:
>  it also globally invalidates the previous GL shader program object across all users of this shader

what do you mean? copies of the same ofShader?

> I wonder if it wasn't possible to make these methods only callable before calling setup()

I think is useful to allow to call them afterwards. it allows to test things easily but perhaps we could have a public recompile method or something like that that needs to be called afterwards, that would also allow to change several at the same time before recompiling but probably that would be confusing and people will forget to call recompile

changing the name to something more explicit is a good idea, usually you will change this from a gui or something so you mostly change them one by one. perhaps setConstant*AndRecompile? :)